src/license_expression/__init__.py

```diff
@@ -116,6 +126,49 @@ class ExpressionParseError(ParseError, ExpressionError):
 ).finditer
 
 
+class ExpressionInfo:
+    """
+    The ExpressionInfo class is returned by Licensing.validate() where it stores
+    information about a given license expression passed into
+    Licensing.validate().
+
+    The ExpressionInfo class has the following fields:
+    - original_license_expression: str.
+      - This is the license expression that was originally passed into Licensing.validate()
+    - normalized_license_expression: str.
+      - If a valid license expression has been passed into `validate()`,
+        then the license expression string will be set in this field.
+    - errors: list
+      - If there were errors validating a license expression,
+        the error messages will be appended here.
+    - valid_symbols: list
+      - If a valid license expression has been passed into `validate()`,
+        then the license symbols from the license expression will be
+        appended here.
+    - valid_exception_symbols: list
+      - If a license symbol in the license expression is a license exception,
+        then that license symbol will be appended here.
+    - invalid_symbols: list
+      - If an invalid license expression has been passed into `validate()`,
+        then the invalid license symbols from the license expression will be
+        appended here.
+    """
+    def __init__(
+            self,
+            original_license_expression,
+            normalized_license_expression=None,
+            errors=None,
+            valid_symbols=None,
+            valid_exception_symbols=None,
+            invalid_symbols=None):
+        self.original_license_expression = original_license_expression
+        self.normalized_license_expression = normalized_license_expression or ''
+        self.errors = errors or []
+        self.valid_symbols = valid_symbols or []
+        self.valid_exception_symbols = valid_exception_symbols or []
+        self.invalid_symbols = invalid_symbols or []
+
+
```

Comment:
A representation helped me play with the code:
```suggestion
    def __repr__(self):
        return (
            'ExpressionInfo(\n'
                f'    original_license_expression={self.original_license_expression!r},\n'
                f'    normalized_license_expression={self.normalized_license_expression!r},\n'
                f'    errors={self.errors!r},\n'
                f'    valid_symbols={self.valid_symbols!r},\n'
                f'    valid_exception_symbols={self.valid_exception_symbols!r},\n'
                f'    invalid_symbols={self.invalid_symbols!r}\n'
            ')'
        )
```

and I found a bug. The valid_exception_symbols is not set correctly: 
```
>>> from license_expression import *
>>> lic = build_spdx_licensing(get_license_index())
>>> lic.validate('Apache-2.0 WITH MIT', strict=True)
ExpressionInfo(
    original_license_expression='Apache-2.0 WITH MIT',
    normalized_license_expression='Apache-2.0 WITH MIT',
    errors=['A plain license symbol cannot be used as an exception in a "WITH symbol" statement. for token: "MIT" at position: 16'],
    valid_symbols=['Apache-2.0 WITH MIT'],
    valid_exception_symbols=['Apache-2.0 WITH MIT'],<===== This is not right.
    invalid_symbols=['MIT']
)
>>> lic.validate('Apache-2.0 WITH MIT', strict=False)
ExpressionInfo(
    original_license_expression='Apache-2.0 WITH MIT',
    normalized_license_expression='Apache-2.0 WITH MIT',
    errors=[],
    valid_symbols=['Apache-2.0 WITH MIT'],
    valid_exception_symbols=['Apache-2.0 WITH MIT'], <===== This is not right. Should be only MIT
    invalid_symbols=[]
)
```
IMHO we should have either a list of invalid exception symbols. Or may be simpler only a list of valid and invalid symbols (not distinguishing what is plain vs. exception.

Also what about making the attribute names shorter since we are in the context of an Expression already?

- original_license_expression ==> original
- normalized_license_expression ==> normalized
- and keep as-is errors, valid_symbols, invalid_symbols?

Reply:
I ended up keeping the following fields:

`original_expression`
`normalized_expression`
`errors`
`invalid_keys`